.github/actions/check-image-and-changes/action.yaml

```diff
@@ -26,103 +26,167 @@ runs:
       id: changes
       shell: bash
       run: |
-        # Ensure that SHORT_SHA is available for subsequent steps in this context
-        echo "SHORT_SHA=${SHORT_SHA:-${{ steps.get-sha.outputs.short_sha }}}" >> $GITHUB_ENV
-
-        # Get base commit for comparison
-        BASE_COMMIT=$(git merge-base ${{ github.event.pull_request.base.sha }} HEAD)
-        echo "BASE_COMMIT=$BASE_COMMIT"
-
-        # Get list of changed files
-        CHANGED_FILES=$(git diff --name-only "$BASE_COMMIT" HEAD)
-        echo "Changed files:"
-        echo "$CHANGED_FILES"
-
-        # Check for [skip-build] tag in commit messages
-        SKIP_BUILD_TAG=false
-        COMMIT_MESSAGES=$(git log --format=%B "$BASE_COMMIT"..HEAD)
-        if echo "$COMMIT_MESSAGES" | grep -q "\[skip-build\]"; then
-          echo "Found [skip-build] tag in commit messages."
-          SKIP_BUILD_TAG=true
+        echo "🔍 Checking changes and build requirements..."
+        
+        # Function to check if file is in always-skip paths
+        check_always_skip_paths() {
+          local file=$1
+          if [[ "$file" == docs/* ]] || [[ "$file" == *.md ]]; then
+            return 0  # true - always skip
+          fi
+          return 1  # false - don't skip automatically
+        }
+        
+        # Function to check if file is in image-dependent paths
+        check_image_dependent_paths() {
+          local file=$1
+          if [[ "$file" == e2e/* ]] || [[ "$file" == e2e-tests/* ]] || [[ "$file" == .ibm/* ]] || [[ "$file" == .rhdh/docker/* ]]; then
+            return 0  # true - may need image
+          fi
+          return 1  # false - doesn't need image
+        }
+        
+        # Function to check if file is relevant code for build
+        check_relevant_code() {
+          local file=$1
+          # Files that affect application build
+          if [[ "$file" == packages/* ]] || [[ "$file" == plugins/* ]] || [[ "$file" == dynamic-plugins/* ]] || [[ "$file" == app-config*.yaml ]] || [[ "$file" == catalog-entities/* ]] || [[ "$file" == docker/* ]] || [[ "$file" == Dockerfile* ]] || [[ "$file" == package*.json ]] || [[ "$file" == yarn.lock ]] || [[ "$file" == tsconfig*.json ]]; then
```

Comment:
eww. this should be a regex
or maybe read in a file that lists the skips and then processes them in a for-loop